homeassistant/loader.py

```diff
@@ -210,6 +211,20 @@ async def async_get_config_flows(
     return flows
 
 
+async def async_get_application_credentials(hass: HomeAssistant) -> list[str]:
+    """Return cached list of application credentials."""
+    integrations = await async_get_custom_components(hass)
+
+    return [
+        *APPLICATION_CREDENTIALS,
```

Comment:
Do you know why we use the generated files at all instead of just looking at the dependencies? Is it because the list of integrations is not complete?

Reply:
great question. In this case it may not be needed, it is for other use cases like config flow, discovery, etc. But even for config flow you could check the config_flow key in the manifest

Reply:
OK. This looks good to me from an application credentials perspective, though I am not familiar with loader.

Reply:
`integrations` is only custom components, so this is ok.